src/coreclr/tools/Common/TypeSystem/Ecma/EcmaModule.cs

```diff
@@ -282,47 +282,57 @@ public bool IsPlatformNeutral
 
         public sealed override MetadataType GetType(string nameSpace, string name, NotFoundBehavior notFoundBehavior)
         {
-            var stringComparer = _metadataReader.StringComparer;
-
-            // TODO: More efficient implementation?
-            foreach (var typeDefinitionHandle in _metadataReader.TypeDefinitions)
+            var currentModule = this;
+            for (int typeForwardingChainSize = 0; typeForwardingChainSize <= 1024; typeForwardingChainSize++)
```

Comment:
This could use a comment - it's not apparent we're trying to detect a loop and why we chose 1024.